Tests/TaskTests/TaskTests.swift

```diff
@@ -266,5 +266,21 @@ class TaskTests: CustomExecutorTestCase {
         waitForExpectations()
         assertExecutorCalled(atLeast: 1)
     }
+    
+    func testThatRecoverAlsoProducesANewTask() {
+        let expectation = self.expectation(description: "recover produces a new task")
+        let task: Task<Int> = anyFailedTask.recoverWithNewTask(upon: executor) { _ in
+            return self.anyFinishedTask
+        }
+        
+        task.upon {
+            XCTAssertEqual($0.value, 42)
```

Comment:
The "42" kind of comes out of nowhere and leaves me confused when reading the test. Maybe assert `$0.value` == `anyFinishedTask.value` to make the connection clear?